(EventTarget) Add an example with multiple `options` set (#17843)
mdn/content: files/en-us/web/api/eventtarget/addeventlistener/index.md

good idea to ensure that the user's browser supports it, since these are an addition
that not all browsers have supported historically. See [Safely detecting option support](#safely_detecting_option_support) for details.

### Event listener with multiple options

The `options` parameter may have multiple properties set.

#### HTML

```html
<button id="example-button">You have not clicked this button.</button>
```

#### JavaScript

```js
const button = document.getElementById("example-button");

> button.addEventListener(
>   "click",
>   (_event) => {
>     button.textContent = "You have clicked this button.";
>   }, {
>     capture: false,
>     passive: true,
>     once: true
>   }
> );

Reviewer: I would format the event handler differently, and since `capture` defaults to `false`, I'm not sure why we are setting it here.

```suggestion
button.addEventListener(
  "click",
  () => button.textContent = "You have clicked this button.",
  {
    passive: true,
    once: true
  }
);
```

Reviewer: I like it when a user can run a live sample more than once, without having to reload the page. Sometimes this means having a separate "Reload" or "Reset" button that reloads the iframe. There might be other ways of accomplishing this here.

But, I see the same complaint could be made about several other examples in this page, so take this as an optional suggestion.

Reviewer 2: Given the patch is about to add this example, let's add an example that can be run several times here. If it was a side fix, I'll agree to move this to a separate PR, but this is the main goal of this PR.

```

> Take notice that multiple properties have values set. In this particular example,
> the event handler function will not be called more than once, *and* it may not
> call {{domxref("Event.preventDefault", "preventDefault()")}} on its event argument.

Reviewer: I'd prefer for the description to be at the start of the example, just after the title. We could amalgamate it with " The `options` parameter may have multiple properties set.", and I might reword things a bit:

```suggestion
You can set more than one of the options in the `options` property. In the following example we are setting two options:
- `passive`, to assert that the handler will not call {{domxref("Event.preventDefault", "preventDefault()")}} 
- `once`, to assert that the event will only be fired once.
```

good idea to ensure that the user's browser supports it, since these are an addition
that not all browsers have supported historically. See [Safely detecting option support](#safely_detecting_option_support) for details.

### Event listener with multiple options

You can set more than one of the options in the `options` parameter. In the following example we are setting two options:
- `passive`, to assert that the handler will not call {{domxref("Event.preventDefault", "preventDefault()")}} 
- `once`, to ensure that the event handler will only be called once.

#### HTML

```html
<button id="example-button">You have not clicked this button.</button>

> <button id="reset-button">Click this button to reset the above button.</button>

Reviewer 2: It isn't above for me:
```suggestion
<button id="reset-button">Click this button to reset the first button.</button>
```

Reviewer 2: ![Capture d’écran 2022-07-07 à 08 55 18](https://user-images.githubusercontent.com/1466293/177710853-46e8d301-a25c-429f-8a13-be35805b50dd.png)

In my case, both buttons are side by side (default MDN dimension, just on a large screen)

```

#### JavaScript

```js
const buttonToBeClicked = document.getElementById("example-button");

const resetButton = document.getElementById("reset-button");

// the text that the button is initalized with
const intialText = buttonToBeClicked.textContent;

// the text that the button contains after being clicked
const clickedText = "You have clicked this button.";

// we hoist the event listener callback function
// to prevent having duplicate listeners attached
function eventListener() {
  buttonToBeClicked.textContent = clickedText;
}

function addListener() {
  buttonToBeClicked.addEventListener(
    "click",
    eventListener,
    {
      passive: true,
      once: true
    }
  );
}

// when the reset button is clicked, the example button is reset,
// and allowed to have its state updated again
resetButton.addEventListener(
  "click",
  () => {

>     button.textContent = intialText;

Reviewer 2: ```suggestion
    buttonToBeClicked.textContent = intialText;
```
